nautobot_ssot/metrics.py

```diff
@@ -0,0 +1,156 @@
+
+
+def metric_ssot_jobs():
+    """Extracts duration of latest SSoT Job run.
+
+    Yields:
+        GaugeMetricFamily: Prometheus Metrics
+    """
+    ssot_job_durations = GaugeMetricFamily(
+        "nautobot_ssot_duration_microseconds",
+        "Nautobot SSoT Job Phase Duration in microseconds",
+        labels=["phase", "job"],
+    )
+
+    for job in Job.objects.filter(slug__icontains="ssot"):
+        last_job_sync = Sync.objects.filter(job_result__job_model_id=job.id).last()
+        if last_job_sync:
+            if last_job_sync.source_load_time:
+                ssot_job_durations.add_metric(
+                    labels=["source_load_time", job.slug],
+                    value=(last_job_sync.source_load_time.seconds * 100000)
+                    + last_job_sync.source_load_time.microseconds,
+                )
+
+            if last_job_sync.target_load_time:
+                ssot_job_durations.add_metric(
+                    labels=["target_load_time", job.slug],
+                    value=(last_job_sync.target_load_time.seconds * 1000000)
+                    + last_job_sync.target_load_time.microseconds,
+                )
+
+            if last_job_sync.diff_time:
+                ssot_job_durations.add_metric(
+                    labels=["diff_time", job.slug],
+                    value=(last_job_sync.diff_time.seconds * 1000000) + last_job_sync.diff_time.microseconds,
+                )
+
+            if last_job_sync.sync_time:
+                ssot_job_durations.add_metric(
+                    labels=["sync_time", job.slug],
+                    value=(last_job_sync.sync_time.seconds * 1000000) + last_job_sync.sync_time.microseconds,
+                )
+
+            if last_job_sync.duration:
+                ssot_job_durations.add_metric(
+                    labels=["sync_duration", job.slug],
+                    value=(last_job_sync.duration.seconds * 1000000) + last_job_sync.duration.microseconds,
+                )
+
+    yield ssot_job_durations
+
+
+def metric_syncs():
+    """Calculate total number of Syncs.
+
+    Yields:
+        GaugeMetricFamily: Prometheus Metrics
+    """
+    sync_gauge = GaugeMetricFamily("nautobot_ssot_sync_totals", "Nautobot SSoT Sync Totals", labels=["sync_type"])
+
+    sync_gauge.add_metric(labels=["total_syncs"], value=Sync.objects.all().count())
+
+    for status_type in ("completed", "failed", "errored", "pending", "running"):
+        sync_gauge.add_metric(
+            labels=[f"{status_type}_syncs"], value=Sync.objects.filter(job_result__status=status_type).count()
+        )
+
+    yield sync_gauge
+
+
+def metric_sync_operations():
+    """Extracts the diff summary operations from each Job's last Sync.
+
+    Yields:
+        GuageMetricFamily: Prometheus Metrics
+    """
+    sync_ops = GaugeMetricFamily(
+        "nautobot_ssot_sync_operations", "Nautobot SSoT operations by Job", labels=["job", "operation"]
```

Comment:
```suggestion
        "nautobot_ssot_sync_total", "Nautobot SSoT operations by Job", labels=["job", "operation"]
```

Reply:
Again, why `total` here as opposed to `totals`? You've got multiple items here. It's not just one operation total.

Reply:
Same reasoning as above.

Reply:
Corrected.